tests/lib/Search/ResultExtractor/AggregationResultExtractor/TermAggregationKeyMapper/ObjectStateAggregationKeyMapperTest.php

```diff
@@ -58,18 +59,25 @@ private function configureObjectStateService(
             ->willReturn($objectStateGroup);
 
         $expectedObjectStates = [];
-        foreach ($objectStateIdentifiers as $i => $objectStateIdentifier) {
+        $map = [];
+        foreach ($objectStateIdentifiers as $objectStateIdentifier) {
             $objectState = $this->createMock(ObjectState::class);
-
-            $this->objectStateService
-                ->expects(self::at($i + 1))
-                ->method('loadObjectStateByIdentifier')
-                ->with($objectStateGroup, $objectStateIdentifier, [])
-                ->willReturn($objectState);
-
             $expectedObjectStates[] = $objectState;
+            $map[$objectStateIdentifier] = $objectState;
         }
 
+        $this->objectStateService
+            ->method('loadObjectStateByIdentifier')
+            ->willReturnCallback(
+                static function ($group, $identifier, $options) use ($objectStateGroup, $map) {
+                    if ($group === $objectStateGroup && isset($map[$identifier]) && $options === []) {
+                        return $map[$identifier];
+                    }
+
+                    throw new InvalidArgumentException("Unexpected arguments: $identifier");
```

Comment:
Does this on production indeed throw `\InvalidArgumentException` instead of `\Ibexa\...\InvalidArgumentException`?